--- a/drivers/net/ethernet/realtek/Kconfig
+++ b/drivers/net/ethernet/realtek/Kconfig
@@ -114,6 +114,18 @@ config R8169_LEDS
 	  Optional support for controlling the NIC LED's with the netdev
 	  LED trigger.
 
+config R8169SOC
+	tristate "Realtek RTD129x SoC Gigabit Ethernet driver"
+	depends on ARCH_RTD129X || COMPILE_TEST
+	select PHYLIB
+	help
+	  This driver supports the integrated Gigabit Ethernet controller
+	  found in Realtek RTD129x SoCs (RTD1294, RTD1295, RTD1296).
+
+	  To compile this driver as a module, choose M here. The module
+	  will be called r8169soc.
+
+	  If unsure, say N.
+
 endif # NET_VENDOR_REALTEK
 
 config RTASE
--- a/drivers/net/ethernet/realtek/Makefile
+++ b/drivers/net/ethernet/realtek/Makefile
@@ -9,3 +9,4 @@ obj-$(CONFIG_8139TOO) += 8139too.o
 r8169-y += r8169_main.o r8169_firmware.o r8169_phy_config.o
 r8169-$(CONFIG_R8169_LEDS) += r8169_leds.o
 obj-$(CONFIG_R8169) += r8169.o
+obj-$(CONFIG_R8169SOC) += r8169soc.o
 obj-$(CONFIG_RTASE) += rtase/
--- a/drivers/gpio/Kconfig
+++ b/drivers/gpio/Kconfig
@@ -557,6 +557,15 @@ config GPIO_RDC321X
 	  Support for the RDC R321x SoC GPIOs over southbridge
 	  PCI configuration space.
 
+config GPIO_RTD129x
+	tristate "Realtek RTD129x GPIO support"
+	depends on ARCH_RTD129X || COMPILE_TEST
+	select GPIOLIB
+	help
+	  This driver supports the GPIO controller found in Realtek RTD129x SoCs.
+
+	  If unsure, say N.
+
 config GPIO_SODAVILLE
 	bool "Intel Sodaville GPIO support"
 	depends on X86 && OF
--- a/drivers/gpio/Makefile
+++ b/drivers/gpio/Makefile
@@ -129,6 +129,7 @@ obj-$(CONFIG_GPIO_RC5T583)		+= gpio-rc5t583.o
 obj-$(CONFIG_GPIO_RDC321X)		+= gpio-rdc321x.o
 obj-$(CONFIG_GPIO_REALTEK_OTTO)		+= gpio-realtek-otto.o
 obj-$(CONFIG_GPIO_REG)			+= gpio-reg.o
+obj-$(CONFIG_GPIO_RTD129x)		+= gpio-rtd129x.o
 obj-$(CONFIG_GPIO_SAMA5D2_PIOBU)	+= gpio-sama5d2-piobu.o
 obj-$(CONFIG_GPIO_SCH)			+= gpio-sch.o
 obj-$(CONFIG_GPIO_SCH311X)		+= gpio-sch311x.o
